solutions/swb-reference/src/templates/onboard-account-tgw.cfn.yaml

```diff
@@ -0,0 +1,777 @@
+            Effect: 'Allow'
+            Principal:
+              AWS: '*'
+            Action:
+              - 'kms:DescribeKey'
+              - 'kms:Encrypt'
+              - 'kms:Decrypt'
+              - 'kms:ReEncrypt*'
+              - 'kms:GenerateDataKey'
+              - 'kms:GenerateDataKeyWithoutPlaintext'
+              - 'kms:CreateGrant'
+              - 'kms:RevokeGrant'
+            Resource: '*'
+            Condition:
+              StringEquals:
+                kms:CallerAccount: !Ref 'AWS::AccountId'
+    Metadata:
+      cfn_nag:
+        rules_to_suppress:
+          - id: F76
+            reason: "There is a condition so that the only caller that can use this KMS key is the aws account. The syntax for the Principal element does not provide a way to specify all identities in an AWS account. But you can achieve this effect by combining the condition key kms:CallerAccount with a Principal element that specifies all AWS identities."
+
+  EncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Join ['', ['alias/', Ref: Namespace, '-encryption-key']]
+      TargetKeyId: !Ref EncryptionKey
+
+  StateChangeEventRuleMainRouting:
+    Type: 'AWS::Events::Rule'
+    Properties:
+      Description: Routes environment state changes to the main event bus
+      EventBusName: !Sub 'arn:aws:events:${AWS::Region}:${AWS::AccountId}:event-bus/default'
+      State: ENABLED
+      EventPattern:
+        account:
+          - !Ref 'AWS::AccountId'
+        detail-type:
+          - anything-but: 'AWS API Call via CloudTrail'
+        source:
+          - anything-but:
+            - 'aws-config'
+            - 'aws-cloudtrail'
+            - 'aws-ssm'
+            - 'aws-tag'
+      Targets:
+        - Arn: !Sub 'arn:aws:events:${AWS::Region}:${MainAccountId}:event-bus/default'
+          Id: 'MainAccountRoutingRule'
+          RoleArn: !GetAtt 
+            - StateChangeEventBridgeIAMrole
+            - Arn
+
+  StateChangeEventBridgeIAMrole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: events.amazonaws.com
+            Action: 'sts:AssumeRole'
+          - Effect: Allow
+            Principal:
+              Service: sagemaker.amazonaws.com
+            Action: 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: PutEventsDestinationBus
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'events:Put*'
+                Resource:
+                  - !Sub 'arn:aws:events:${AWS::Region}:${MainAccountId}:event-bus/default'
+
+
+  GatewayLoadBalancerSubnet1:
+    Condition: PublicVPC
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref "VPC"
+      CidrBlock: !Ref "GLBSubnet1CIDR"
+      AvailabilityZone: "us-east-1a"
```

Comment:
Can we customize this? Not all SWB deployments will be in this region.

Reply:
I have making this generic as an AC in the follow up story, but I can do it here if you like

Reply:
Nope. Can wait for your upcoming ticket.

Reply:
Fixed it